python/lsst/cp/pipe/cpFilterScan.py

```diff
@@ -0,0 +1,443 @@
+            wavelengthKey = float(wavelengthKey)
+            self.log.debug(f"Scan: {exposureId} {wavelengthKey} {scan}")
+
+            # Append the scan for this exposure to the list of other
+            # scans taken at this wavelength setting:
+            if wavelengthKey in filterScanResults:
+                filterScanResults[wavelengthKey].append(scan)
+            else:
+                filterScanResults[wavelengthKey] = [scan]
+
+        filterScan = []
+        for wavelength in filterScanResults.keys():
+            # We expect there to be at least one pair of exposures
+            # with a given wavelength setting: One with a filter in
+            # the beam, and one with the "reference filter" (which
+            # should be the empty/white/no-filter setting).  The ratio
+            # of the scaled fluxes in this pair should give the filter
+            # transmission at that wavelength for that filter.
+            scans = filterScanResults[wavelength]
+
+            referenceScan = [x for x in scans if x['physical_filter'] == self.config.referenceFilter]
+            if len(referenceScan) == 0:
+                # No reference scan at this wavelength.
+                self.log.warning(f"No reference scan at this wavelength: {wavelengthKey}")
+                continue
+            referenceScan = referenceScan[0]
+            referenceValue = referenceScan['scale'] / referenceScan['flux']
+
+            # Create a dictionary of measurements at this wavelength,
+            # using the filter names as the keys.  Since we may do
+            # multiple filters in a single sequence (iterating through
+            # all filters at a certain monochromator setting),
+            # initialize all known filters (from our initial exposure
+            # scan) to NaN.  Note that the reference filter is a known
+            # filter, and should have a throughput of 1.0 relative to
+            # itself.
+            wavelengthScan = {'wavelength': float(wavelength)}
+            for filterName in filterSet:
+                wavelengthScan[filterName] = np.nan
+
+            for scan in scans:
+                # If the entry for this filter already exists, then we
+                # have multiple entries at this wavelength.
+                if np.isfinite(wavelengthScan[scan['physical_filter']]):
+                    self.log.warning(
+                        f"Multiple instances of filter {scan['physical_filter']} at {wavelength}"
+                    )
+
+                wavelengthScan[scan['physical_filter']] = referenceValue * scan['flux'] / scan['scale']
+
+            filterScan.append(copy.copy(wavelengthScan))
+
+        outMetadata = PropertyList()
+        self.combine.combineHeaders(inputExpHandles, calib=None, metadata=outMetadata,
+                                    calibType="FilterScan", scales=None)
+        filteredMetadata = {k: v for k, v in outMetadata.toDict().items() if v is not None}
+
```

Comment:
Adding a comment here for what fields we will have in the final `filterScan` would be really useful for the future/downstream processing.